packages/extension-starter/src/setup-app.ts

```diff
@@ -79,14 +79,18 @@ class Sample extends ApplicationExtension<Config> {
      *
      * NOTE: If you instead want to modify a list of all the routes, refer to the `beforeRouteMatch` below.
      */
-    getRoutes(params: GetRoutesParams): RouteProps[] {
-        return [
+    getRoutesAsync(params: GetRoutesParams): Promise<RouteProps[]> {
+        return Promise.resolve([
             {
                 exact: true,
                 path: this.getConfig().path,
                 component: SamplePage
             }
-        ]
+        ])
+    }
+
+    getComponentMap() {
+        return {SamplePage}
```

Comment:
to ensure that the solution works for extensions that implement `getRoutesAsync` I have temporarily updated `extension-starter`

Reply:
`getComponentMap` is required because it is used to deserialize the serialized routes from the server in the client.

Extensions need to have this implemented so that we can map a `componentName` to `component` during deserialization

Reply:
Thanks for making this temporary change to `extension-starter`. I found myself so focused on the SEO extension, that I forgot the possibly-more-common case, which the extension-starter is in.